src/EditorFeatures/CSharpTest/Classification/SyntacticClassifierTests.cs

```diff
@@ -3605,5 +3605,48 @@ public void LoadDirective()
                 PPKeyword("load"),
                 String("\"file.csx\""));
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Classification)]
+        public void IncompleteAwaitInNonAsyncContext()
+        {
+            var code = @"
+void M()
+{
+    var x = await
+}";
+            TestInClass(code,
+                Keyword("void"),
+                Identifier("M"),
+                Punctuation.OpenParen,
+                Punctuation.CloseParen,
+                Punctuation.OpenCurly,
+                Keyword("var"),
+                Identifier("x"),
+                Operators.Equals,
+                Keyword("await"),
+                Punctuation.CloseCurly);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Classification)]
+        public void CompleteAwaitInNonAsyncContext()
+        {
+            var code = @"
+void M()
+{
+    var x = await;
```

Comment:
So in the latter two cases this approach will work.  It won't work for the ```M(await)``` case.  I could go with the approach I took with VB and just try to bind it semantically.  If it fails, I can just assume it had to be a keyword...  I wasn't thinking it had to be so 'perfect'.  I was happy with something that would get it right 99/100 times. :)